### PR TITLE
Fix react npm dependency & rollup config

### LINKED ISSUE #107: React dependency
It seems there is an issue with the React dependency `"react": "file:../node_modules/react"`.
Shouldn't there be a React version number instead of local files?

### PR DESCRIPTION
**Pull request by haikov**
This PR fixes #107 by setting the same react dep. version as in the other package (`latest`).

I also updated the rollup config file, since the one in main branch throws the following warning and doesn't write anything in `dist` folder.

I tested the build generated from this branch and it seems to be mostly working (I'm on 11-rc version of realm and I had to comment `throw new Error('You need to be on realm 10.20.0 or higher');` line, but I suspect it's some API change?

```
(!) You have passed an unrecognized option
Unknown output options: 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14. Allowed options: amd, assetFileNames ....
```

### CODE REVIEW

**Comment by kneth** [comment]
@haikov Thank you for the pull request. I am assigning a reviewer to move the process forward.

**Comment by akashraj9828** [comment]
Facing the same issue on my machine. @haikov @kneth @kraenhansen Can u please test this PR and merge it as soon as possible? Thanks

File: realm-flipper-plugin-device/package.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "react": ">=17"
   },
   "dependencies": {
-    "react": "file:../node_modules/react",
+    "react": "latest",
```
Comment (kraenhansen) [inline realm-flipper-plugin-device/package.json:42]: I'd rather we pin this to a specific major version.
```suggestion
    "react": "^18.2.0",
```

**Comment by Steven-Thelin** [comment]
This is still an issue and it's not possible to install the debugger.
@haikov Would you mind checking the suggestion by @kraenhansen?

**Comment by haikov** [comment]
@cla-bot check

**Comment by haikov** [comment]
@kraenhansen I've updated the PR with explicit version of React. Should be ready to merge now.

**Comment by hadnet** [comment]
I'm trying to use the realm-flipper-plugin-device@1.0.11 📦 but I get these errors

(1)

> :x: 
> realm-flipper-plugin-device@1.0.10 build
> rollup -c



(2)
When commenting out lines above and trying again I get this error when trying to use it with useRealm()

>:x:
>[Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) >but got: object.
>
>Check the render method of `ComponentScreen`.

Just to clarify I'm using it exactly like in de docs but with useRealm:
```ts
import RealmPlugin from 'realm-flipper-plugin-device';

MyComp() {

  const realm = useRealm()
  
  return (
     <View>
        <RealmPlugin realms={[realm]} /> {/* pass in the Realm to the plugin*/}
        // ...
     </View>
  );

}
```

**Comment by takameyer** [comment]
@hadnet Can you please open a new issue, including what version of Realm you are using?  Also I have just released `1.0.27`, so I recommend updating to make sure we can be sure the issue isn't already fixed.

**Comment by hadnet** [comment]
> @hadnet Can you please open a new issue, including what version of Realm you are using? Also I have just released `1.0.27`, so I recommend updating to make sure we can be sure the issue isn't already fixed.

Ok, installed 1.0.27 but now I got this new error:
Screenshot 2022-12-02 at 6 32 53 PM

I'm using:
```json
{
    "realm": "^11.3.0",
    "@realm/react": "^0.4.1"
}
```

**Comment by takameyer** [comment]
@hadnet at a glance, it seems that you may have encountered a dependency conflict.  Can you share the contents of your `package.json`?

**Comment by hadnet** [comment]
it's a simpler RN boilerplate with this packages

```json
  "dependencies": {
    "@hadnet/react-native-skeleton": "^1.0.0-alpha.1",
    "@react-native-masked-view/masked-view": "^0.2.7",
    "@realm/react": "^0.4.1",
    "iconsax-react-native": "^0.0.8",
    "moti": "^0.18.0",
    "react": "18.1.0",
    "react-native": "0.70.4",
    "react-native-gesture-handler": "^2.6.1",
    "react-native-get-random-values": "^1.8.0",
    "react-native-linear-gradient": "^2.6.2",
    "react-native-reanimated": "^3.0.0-rc.2",
    "react-native-shimmer": "^0.6.0",
    "react-native-skeleton-placeholder": "^5.0.0",
    "react-native-svg": "^13.4.0",
    "realm": "^11.3.0"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@babel/plugin-proposal-decorators": "^7.20.5",
    "@babel/runtime": "^7.12.5",
    "@react-native-community/eslint-config": "^2.0.0",
    "@realm/babel-plugin": "^0.1.0",
    "@tsconfig/react-native": "^2.0.2",
    "@types/jest": "^26.0.23",
    "@types/react-native": "^0.70.0",
    "@types/react-test-renderer": "^18.0.0",
    "@typescript-eslint/eslint-plugin": "^5.37.0",
    "@typescript-eslint/parser": "^5.37.0",
    "babel-jest": "^26.6.3",
    "eslint": "^7.32.0",
    "jest": "^26.6.3",
    "metro-react-native-babel-preset": "^0.72.1",
    "react-native-builder-bob": "^0.18.3",
    "react-native-flipper": "^0.176.0",
    "react-native-flipper-performance-plugin": "^0.3.1",
    "react-test-renderer": "18.1.0",
    "realm-flipper-plugin-device": "^1.0.27",
    "typescript": "^4.8.3"
  },
```

**Pull request merged**